src/Agent/NewRelic/Agent/Core/WireModels/LoadedModuleWireModelCollection.cs

```diff
@@ -12,9 +16,206 @@ public class LoadedModuleWireModelCollection
+                {
+                    loadedModule.Data.Add("sha1Checksum", sha1FileHash);
+                    loadedModule.Data.Add("sha512Checksum", sha512FileHash);
+                }
+
+                if (TryGetAssemblyHashCode(assembly, out var assemblyHashCode))
+                {
+                    loadedModule.Data.Add("assemblyHashCode", assemblyHashCode);
+                }
+                if (TryGetCompanyName(assembly, out var companyName))
+                {
+                    loadedModule.Data.Add("Implementation-Vendor", companyName);
+                }
+                if (TryGetCopyright(assembly, out var copyright))
+                {
+                    loadedModule.Data.Add("copyright", copyright);
+                }
+
+                // Use the .Name here and in GetLoadedModules
+                loadedModulesCollection.LoadedModules.Add(loadedModule);
+            }
+
+            return loadedModulesCollection;
+        }
+
+        private static bool TryGetAssemblyName(Assembly assembly, out string assemblyName)
+        {
+            try
+            {
+                if (assembly.IsDynamic)
+                {
+                    assemblyName = assembly.GetName().Name;
+                }
+                else
+                {
+                    assemblyName = Path.GetFileName(assembly.Location);
+                }
+
+                if (string.IsNullOrWhiteSpace(assemblyName))
+                {
+                    return false;
+                }
+
+                return true;
+            }
+            catch
+            {
+                assemblyName = null;
+                return false;
+            }
+        }
+
+        private static bool TryGetPublicKeyToken(AssemblyName assemblyDetails, out string publicKey)
+        {
+            try
+            {
+                publicKey = BitConverter.ToString(assemblyDetails.GetPublicKeyToken()).Replace("-", "");
+                if (string.IsNullOrWhiteSpace(publicKey))
+                {
+                    return false;
+                }
+
+                return true;
+            }
+            catch
+            {
+                publicKey = null;
+                return false;
+            }
+        }
+
+        private static bool TryGetShaFileHashes(Assembly assembly, out string sha1FileHash, out string sha512FileHash)
+        {
+            try
+            {
+                var location = assembly.Location;
+                if (string.IsNullOrEmpty(location))
+                {
+                    sha1FileHash = null;
+                    sha512FileHash = null;
+                    return false;
+                }
+
+                if (!File.Exists(location))
+                {
+                    sha1FileHash = null;
+                    sha512FileHash = null;
+                    return false;
+                }
+
+                using (var stream = File.OpenRead(location))
```

Comment:
You might want to consider calculating both hashes in a single pass using [this](https://stackoverflow.com/questions/2124468/possible-to-calculate-md5-or-other-hash-with-buffered-reads/2124500#2124500) approach.

Reply:
I refactored and using that as a base, made it single pass.  Let me know what you think!

Reply:
Yep, that's just what I was thinking, thanks! Maybe just a short unit test to make sure the hashes are correct?